spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/lock/JdbcLockRegistry.java

```diff
@@ -305,12 +307,21 @@ public void unlock() {
 			try {
 				while (true) {
 					try {
-						this.mutex.delete(this.path);
-						return;
+						if (this.mutex.delete(this.path)) {
+							return;
+						}
+						else {
+							throw new ConcurrentModificationException();
+							// the lock is no longer owned by current process, the exception should be handle and rollback the execution result
+						}
 					}
 					catch (TransientDataAccessException | TransactionTimedOutException | TransactionSystemException e) {
 						// try again
 					}
+					catch (ConcurrentModificationException e) {
+						throw new ConcurrentModificationException("Lock was released in the store due to expiration. " +
```

Comment:
I'm not sure what is the reason to catch it here and re-throw immediately.
Why we just cannot include the message in the original one?

Reply:
There is a catch block for catching general `Exception` in this method which will covert exceptions to `DataAccessResourceFailureException`.
To avoid `ConcurrentModificationException` be convert to `DataAccessResourceFailureException`, it needs to has it's own catch block.
Yes, we can include the message in the original one.

Reply:
Yeah... I see.
OK.
Let it be then like:
```
catch (ConcurrentModificationException e) {
    throw e;
}
```
And move that message to the original `throw` point.
With this `throw e;` we just don't create a new object.